dicoogle/src/main/java/pt/ua/dicoogle/taskManager/RunningIndexTasks.java

```diff
@@ -54,7 +55,7 @@ public static RunningIndexTasks getInstance() {
 	}
 
 	public RunningIndexTasks() {
-		taskRunningList = new LinkedHashMap<>();
+		taskRunningList = new ConcurrentHashMap<>();
```

Comment:
We may need to benchmark this, but there are likely better defaults for these parameters. `16` for the concurrency level is too high. I would try these:

```suggestion
		taskRunningList = new ConcurrentHashMap<>(65_536, 0.75, 4);
```

(increased the initial capacity because resizing the table may be a bit more expensive than with a plain hash map)